rush/rush-lib/src/data/RushConfigurationProject.ts

```diff
@@ -97,6 +99,75 @@ export default class RushConfigurationProject {
   }
 
   /**
+   * Generate a temp_module package.json for this project based on an overall configuration
+   */
+  public generateTempModule(rushConfiguration: RushConfiguration): IPackageJson {
```

Comment:
Why are you moving this logic into "rush-lib"?  The RushConfiguration classes are about parsing config files, not performing file system actions.  Also, all the code that cares about generating temp modules is in the "rush" project.